Add module project_task_link

ddufresne opened the pull request:
https://www.numigi.com/web#id=17193&action=298&model=project.task&view_type=form&menu_id=200

ddufresne committed:
Add module project_task_link

ddufresne committed:
Add module project_task_link

ddufresne commented on project_task_link/__init__.py:
```diff
@@ -0,0 +1,4 @@
+# © 2019 Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from . import models
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['.models' imported but unused](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4478774)

ddufresne commented on project_task_link/models/common.py:
```diff
@@ -0,0 +1,80 @@
+DEFAULT_TASK_REF_FORMAT = "TA#{task_id}"
+
+
+def _get_task_ref_regex(env: 'Environment') -> str:
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [invalid syntax](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4478774)

ddufresne commented on project_task_link/static/src/js/html_field.js:
```diff
@@ -0,0 +1,48 @@
+odoo.define("project_task_link.html_field", function(require) {
+"use strict";
+
+var HtmlField = require('web_editor.backend').FieldTextHtmlSimple;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Strings must use doublequote. (quotes)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4478774)

ddufresne commented on project_task_link/models/__init__.py:
```diff
@@ -0,0 +1,4 @@
+# © 2019 Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from . import project_task
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['.project_task' imported but unused](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4478774)

ddufresne commented on project_task_link/static/src/js/html_field.js:
```diff
@@ -0,0 +1,48 @@
+    _renderReadonly(){
+        this._super.apply(this, arguments);
+        var self = this;
+        var taskLinks = this.$('a').filter(function() {
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Strings must use doublequote. (quotes)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4478774)

ddufresne commented on project_task_link/static/src/js/html_field.js:
```diff
@@ -0,0 +1,48 @@
+    © 2019 Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+    License LGPL-3.0 or later (http://www.gnu.org/licenses/LGPL.html).
+*/
+odoo.define("project_task_link.html_field", function(require) {
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['odoo' is not defined. (no-undef)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4478774)

ddufresne commented on project_task_link/models/common.py:
```diff
@@ -0,0 +1,80 @@
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import re
+from lxml import etree
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Using etree to parse untrusted XML data is known to be vulnerable to XML attacks. Replace etree with the equivalent defusedxml package.](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4478774)

ddufresne commented on project_task_link/static/src/js/html_field.js:
```diff
@@ -0,0 +1,48 @@
+            method: "get_formview_action",
+            args: [[taskId]],
+            params: {
+                context: odoo.session_info.user_context,
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['odoo' is not defined. (no-undef)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4478774)

ddufresne committed:
Codacy

foutoucour commented on project_task_link/README.rst:
```diff
@@ -0,0 +1,67 @@
+Project Task Link
+=================
+This module allows to insert dynamic links in the descriptions of tasks.
```
```suggestion
This module allows to insert dynamic links in the description of tasks.
```

foutoucour commented on project_task_link/tests/test_project_task.py:
```diff
@@ -0,0 +1,103 @@
+        assert self.task_link not in self.task.description
+
+    @data(
+        "TA{}",
```
```suggestion
        "TA{}",
        "TA#{}",
```

foutoucour commented on project_task_link/models/common.py:
```diff
@@ -0,0 +1,80 @@
+from lxml import etree
+
+
+DEFAULT_TASK_REF_REGEX = r'[Tt][Aa]\#?\d+'
```
Je suis partagé sur cette implémentation. 
Je suis partagé car nous allons avoir besoin de la même logique pour Github PR à un moment,
et je trouve le tout compliqué.

Je comprends que ça permet l'utilisation d'un ir.config.parameter pour rendre custom la regex, mais est-ce la bonne manière de répondre au besoin, je ne suis pas sûr.

Je te propose un chat de 15-30 mins demain sur le sujet, ok ?

ddufresne committed:
Add module project_task_reference

Separate the logic of searching task based on strings.
This module could be reused later to match a pull request title with a task.

ddufresne committed:
Update README

ddufresne commented on project_task_reference/models/project_task.py:
```diff
@@ -0,0 +1,74 @@
+DEFAULT_TASK_REF_FORMAT = "TA#{id}"
+
+
+def _get_task_ref_regex(env: 'Environment') -> str:
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [invalid syntax](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4478774)

ddufresne replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [invalid syntax](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4478774)

ddufresne replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [invalid syntax](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4478774)

ddufresne commented on project_task_reference/tests/test_project_task.py:
```diff
@@ -0,0 +1,77 @@
+        reference = self.env['project.task']._extract_references_from_text(text)[0]
+        assert reference.normalized_string == "TA#{}".format(self.task_1.id)
+
+    def _make_reference_from_task_id(self, task_id):
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4478774)

ddufresne replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4478774)

ddufresne replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4478774)

ddufresne commented on project_task_reference/__init__.py:
```diff
@@ -0,0 +1,4 @@
+# © 2019 Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from . import models
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['.models' imported but unused](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4478774)

ddufresne replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['.models' imported but unused](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4478774)

ddufresne replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['.models' imported but unused](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4478774)

ddufresne commented on project_task_link/models/common.py:
```diff
@@ -0,0 +1,51 @@
+from typing import List
+
+
+def _find_task_references(env: 'Environment', text: str) -> List[TaskReference]:
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [invalid syntax](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4478774)

ddufresne replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [invalid syntax](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4478774)

ddufresne replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [invalid syntax](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4478774)

ddufresne commented on project_task_reference/reference.py:
```diff
@@ -0,0 +1,33 @@
+
+class TaskReference:
+
+    def __init__(self, values: Mapping[str, str], string: str, format_: str):
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [invalid syntax](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4478774)

ddufresne replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [invalid syntax](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4478774)

ddufresne replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [invalid syntax](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4478774)

ddufresne commented on project_task_reference/models/project_task.py:
```diff
@@ -0,0 +1,74 @@
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import re
+from odoo import api, models, _
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unused api imported from odoo](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4478774)

ddufresne replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unused api imported from odoo](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4478774)

ddufresne replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unused api imported from odoo](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4478774)

foutoucour commented:
@ddufresne ça fail :D

ddufresne committed:
Remove assert

ddufresne committed:
Merge github.com:Numigi/odoo-project-addons into project_task_link

ddufresne committed:
Codacy
